frame/system/src/lib.rs

```diff
@@ -1299,6 +1308,21 @@ impl<T: Trait + Send + Sync> SignedExtension for CheckWeight<T> {
 	) -> TransactionValidity {
 		Self::do_validate(info, len)
 	}
+
+	fn post_dispatch(
+		_pre: Self::Pre,
+		info: Self::DispatchInfo,
+		_len: usize,
+		result: &DispatchResult,
+	) -> Result<(), TransactionValidityError> {
+		// Since mandatory dispatched do not get validated for being overweight, we are sensitive
+		// to them actually being useful. Block producers are thus not allowed to include mandatory
+		// extrinsics that result in error.
+		if info.class == DispatchClass::Mandatory && result.is_err() {
```

Comment:
Shouldn't this be a general case? Aka even without the `CheckWeight` signed extension we should check the result?

Reply:
They are already disallowed via `pre_dispatch`, why is this check needed?

Inherents go through `post_dispatch_unsigned` - I believe that's where it's supposed to be tested?

Reply:
If we're not using `CheckWeight` than `Mandatory` is just a low-priority version of `Operational`, so I don't think we need to handicap it in the general case.